kpi/tests/api/v2/test_api_asset_counts.py

```diff
@@ -0,0 +1,261 @@
+        submissions = [
+            {
+                '__version__': self.asset.latest_deployed_version.uid,
+                'q1': 'a1',
+                '_submitted_by': 'anotheruser',
+                '_submission_time': '2022-09-07T13:21:33',
+            },
+            {
+                '__version__': self.asset.latest_deployed_version.uid,
+                'q1': 'a3',
+                '_submitted_by': '',
+                '_submission_time': '2022-09-12T16:31:33',
+            },
+            {
+                '__version__': self.asset.latest_deployed_version.uid,
+                'q1': 'a1',
+                '_submitted_by': '',
+                '_submission_time': '2022-09-12T17:31:33',
+            },
+        ]
+
+        self.asset.deployment.mock_submissions(submissions)
+        count_url = reverse(
+            self._get_endpoint('asset-counts-list'),
+            kwargs={'parent_lookup_asset': self.asset_uid}
+        )
+        # Test anonymous user
+        self.client.logout()
+        response = self.client.get(count_url)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_count_endpoint_anonymous_user_public_access(self):
+        self.client.login(username='someuser', password='someuser')
+        self.asset.content = {
+            'survey': [
+                {
+                    'type': 'select_one',
+                    'label': 'q1',
+                    'select_from_list_name': 'iu0sl99'
+                },
+            ],
+            'choices': [
+                {'name': 'a1', 'label': ['a1'], 'list_name': 'iu0sl99'},
+                {'name': 'a3', 'label': ['a3'], 'list_name': 'iu0sl99'},
+            ]
+        }
+        self.asset.save()
+        self.asset.deploy(backend='mock', active=True)
+        submissions = [
+            {
+                '__version__': self.asset.latest_deployed_version.uid,
+                'q1': 'a1',
+                '_submitted_by': 'anotheruser',
+                '_submission_time': '2022-09-07T13:21:33',
+            },
+            {
+                '__version__': self.asset.latest_deployed_version.uid,
+                'q1': 'a3',
+                '_submitted_by': '',
+                '_submission_time': '2022-09-12T16:31:33',
+            },
+            {
+                '__version__': self.asset.latest_deployed_version.uid,
+                'q1': 'a1',
+                '_submitted_by': '',
+                '_submission_time': '2022-09-12T17:31:33',
+            },
+        ]
+
+        self.asset.deployment.mock_submissions(submissions)
+        count_url = reverse(
+            self._get_endpoint('asset-counts-list'),
+            kwargs={'parent_lookup_asset': self.asset_uid}
+        )
+        # Test anonymous user with public access to data
+        anonymous_user = User.objects.get(username='AnonymousUser')
+        self.asset.assign_perm(anonymous_user, PERM_VIEW_ASSET)
```

Comment:
`assign_perm` is smart enough to assign implied permissions. `PERM_VIEW_ASSET` is implied when assigning `PERM_VIEW_SUBMISSIONS`.